misc/scripts/prepare-db-upgrade.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/sh
+#
+# Prepare the upgrade and downgrade script directories for a DB Schema upgrade.
+
+set -e
+set -u
+
+app_name="$(basename "$0")"
+
+usage()
+{
+  exit_code="$1"
+  shift
+
+  cat >&2 <<EOF
+${app_name}: $@
+${app_name}: Generate skeleton upgrade script.
+Usage: ${app_name} [--prev_hash <COMMITISH>] [--lang <LANG>]"
+
+--prev-hash <COMMITISH>
+        Hash/branch to use to get SHA1 for previous DB scheme.
+        Default: origin/main
+
+--lang <LANG>
+        Language to update the schema for.
+
+Must be run within the git repo needing an update.
+EOF
+  exit "${exit_code}"
+}
+
+prev_hash="origin/main"
+lang="cpp"
```

Comment:
Could we remove this, and make the `--lang` flag mandatory? If you forget to provide it after making a change to, say, the Ruby dbscheme, it just tells you there's no work to do, because it was looking at the C++ dbscheme.

Reply:
Agreed, and done.